src/parsing/staticallyCompileXPath.ts

```diff
@@ -61,9 +62,7 @@ export default function staticallyCompileXPath(
 		const ast = parseExpression(xpathString, compilationOptions);
 
 		if (compilationOptions.annotateAst) {
-			annotateAst(ast, {
-				staticContext: rootStaticContext,
-			});
+			annotateAst(ast, new AnnotationContext(rootStaticContext));
```

Comment:
Maybe we could reuse the same annotation context in  both calls here?

Reply:
Makes sense